examples/scribe_v1alpha1_replicationsource_restic.yaml

```diff
@@ -0,0 +1,22 @@
+---
+apiVersion: scribe.backube/v1alpha1
+kind: ReplicationSource
+metadata:
+  name: database-source
+  namespace: source
+spec:
+  sourcePVC: mysql-pv-claim
+  trigger:
+    schedule: "*/5 * * * *"
+  restic:
+    pruneIntervalDays: 15
+    repository: restic-config
+    retain:
+      last: 1
+      hourly: 1
+      daily: 1
+      weekly: 1
+      monthly: 1
+      yearly: 1
+      within: "3w15h"
```

Comment:
@JohnStrunk Should `within` be constructed from `hourly`, `daily`, `weekly`, and `yearly` >

Reply:
* I'd suggest `last: 1` if `.spec.retain` is `nil` (i.e., the default).
* I tried to define "within" so that it could be directly passed to restic. See: https://restic.readthedocs.io/en/stable/060_forget.html#removing-snapshots-according-to-a-policy An aside --- While "within" looks like a golang time.Duration, it's not. Restic has created their own definition with longer durations.

Reply:
So `within` provides value for `restic duration` in `--keep-within `. Does that means `within` has a constant value of `3w15h` in `Retain`?

Reply:
Assuming I understand what you're asking, yes...  If `within != nil` then just pass whatever value it has via  `--keep-within ...`

Reply:
All of the items in retain are intended to correspond directly to the corresponding restic cli flag.